core/modules/core/src/com/haulmont/yarg/formatters/impl/HtmlFormatter.java

```diff
@@ -88,60 +93,87 @@ public void setFontsDirectory(String fontsDirectory) {
         this.fontsDirectory = fontsDirectory;
     }
 
+    public void setOpenHtmlForPdfConversion(boolean openHtmlForPdfConversion) {
+        this.openHtmlForPdfConversion = openHtmlForPdfConversion;
+    }
+
     protected void renderPdfDocument(String htmlContent, OutputStream outputStream) {
-        ITextRenderer renderer = new ITextRenderer();
         File temporaryFile = null;
         try {
             temporaryFile = File.createTempFile("htmlReport", ".htm");
             DataOutputStream dataOutputStream = new DataOutputStream(new FileOutputStream(temporaryFile));
-            dataOutputStream.write(htmlContent.getBytes(Charset.forName("UTF-8")));
+            dataOutputStream.write(htmlContent.getBytes(StandardCharsets.UTF_8));
             dataOutputStream.close();
 
-            loadFonts(renderer);
+            PdfConverter converter = createPdfConverter();
+            loadFonts(converter);
 
             String url = temporaryFile.toURI().toURL().toString();
-            renderer.setDocument(url);
+            converter.convert(url, outputStream);
 
-            renderer.layout();
-            renderer.createPDF(outputStream);
         } catch (Exception e) {
             throw wrapWithReportingException("", e);
         } finally {
             FileUtils.deleteQuietly(temporaryFile);
         }
     }
 
+    protected PdfConverter createPdfConverter() {
+        return openHtmlForPdfConversion ? new OpenHtmlToPdfConverter() : new ITextPdfConverter();
+    }
+
```

Comment:
Introduce new factory: HtmlToPdfConverterFactory
Set HtmlToPdfConverterFactory for HTML formatter in the DefaultFormatterFactory.